pallets/identity/src/lib.rs

```diff
@@ -583,6 +580,27 @@ decl_module! {
         pub fn register_custom_claim_type(origin, ty: Vec<u8>) {
             Self::base_register_custom_claim_type(origin, ty)?;
         }
+
+        /// Register `target_account` with a new Identity and issue a CDD claim with a blank CddId
+        ///
+        /// # Failure
+        /// - `origin` has to be a active CDD provider. Inactive CDD providers cannot add new
+        /// claims.
+        /// - `target_account` (primary key of the new Identity) can be linked to just one and only
+        /// one identity.
+        /// - External secondary keys can be linked to just one identity.
+        #[weight = <T as Config>::WeightInfo::cdd_register_did(secondary_keys.len() as u32).saturating_add(<T as Config>::WeightInfo::add_claim())]
+        pub fn cdd_register_did_with_cdd(
+            origin,
+            target_account: T::AccountId,
+            secondary_keys: Vec<SecondaryKey<T::AccountId>>,
+            expiry: Option<T::Moment>
+        ) {
+            let (cdd_did, target_did) = Self::base_cdd_register_did(origin, target_account, secondary_keys)?;
+            let cdd_claim = Claim::CustomerDueDiligence(CddId::default());
+            Self::base_add_claim(target_did, cdd_claim, cdd_did, expiry)?;
```

Comment:
These should be wrapped in `with_transaction`.
```suggestion
            with_transaction(|| {
                let (cdd_did, target_did) = Self::base_cdd_register_did(origin, target_account, secondary_keys)?;
                let cdd_claim = Claim::CustomerDueDiligence(CddId::default());
                Self::base_add_claim(target_did, cdd_claim, cdd_did, expiry)
            })
```

After the Substrate update, this will not be needed.

Reply:
@Neopallium I can do - in this specific case where we know that cdd_claim won't be a custom claim, I can't see any way that base_add_claim can fail though. If you think we need it though I'll add it.

Reply:
Correct.  I didn't look at what checks `base_add_claim` did.